### DISCUSSION TITLE
How to use peerJS API with different signaling service like Firebase?

### DISCUSSION THREAD

**Discussion by jcubic** [peers/peerjs#1049] [Q&A]
PeerJS have really nice API can I use it with service like Firebase to exchange the offers and other WebRTC messages?

Or is forking the project or cloning the API the only way?

**Comment by jonasgloning** [comment]
There’s currently no supported way to do that.
(And at the scale of the public server, it would be prohibitively expensive)

[Sharedrop](https://github.com/szimek/sharedrop) has forked an ancient version of PeerJS and extended it with firebase signaling; maybe you can find some inspiration there.
https://github.com/szimek/sharedrop/blob/master/vendor/peer.js

**Reply by justin0mcateer** [reply]
We have a need to use a different signaling path that the WebSocket that PeerJS uses by default. It seems that making the 'Socket' interface injectable is likely sufficient. Would there by any interest in a PR to make the Socket injectable?

**Reply by jcubic** [reply]
@justin0mcateer If you need a different signaling strategy I suggest using a different library like [feross/simple-peer](https://github.com/feross/simple-peer).

**Reply by jonasgloning** [reply]
@justin0mcateer Yes, why not, please go ahead. This seems like a small change.
